app/controllers/application_controller.rb

```diff
@@ -441,7 +441,7 @@ def accept_key_auth_actions
 
   # Returns a string that can be used as filename value in Content-Disposition header
   def filename_for_content_disposition(name)
-    request.env['HTTP_USER_AGENT'] =~ %r{(MSIE|Trident)} ? ERB::Util.url_encode(name) : name
```

Comment:
Just for context, `match` and `=~` fill [some pseudo global variables](https://docs.ruby-lang.org/en/3.2/Regexp.html#class-Regexp-label-Regexp+Global+Variables) (one of the things that shows Matz's love for Perl). Even though they use the `$` prefix they are thread- and method local. Also, the match object is created.

Using `match?` only checks that the string matches the regex without generating a match object and not filling the pseudo globals